catalog/spatial/geocoding/spatial-geocoding-feature/src/main/java/org/codice/ddf/spatial/geocoding/feature/CatalogFeatureIndexer.java

```diff
@@ -0,0 +1,179 @@
+      throws FeatureIndexingException {
+    try {
+      security.runWithSubjectOrElevate(
+          () -> {
+            Metacard metacard = null;
+
+            if (!create) {
+              metacard = findMetacardForFeature(feature);
+            }
+            if (metacard == null) {
+              metacard = createMetacardForFeature(feature);
+              catalogFramework.create(new CreateRequestImpl(metacard));
+            } else {
+              catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+            }
+            return null;
+          });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private Metacard createMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
+    Metacard metacard = new MetacardImpl();
+
+    String countryCode = (String) feature.getAttribute("ISO_A3");
+    metacard.setAttribute(new AttributeImpl(Metacard.TITLE, countryCode));
+
+    WKTWriter writer = new WKTWriter();
+    String wkt = writer.write((Geometry) feature.getDefaultGeometry());
+    metacard.setAttribute(new AttributeImpl(Metacard.GEOGRAPHY, wkt));
+
+    List<Serializable> tags = new ArrayList<>();
+    tags.add("gazetteer");
+    tags.add("country");
+    metacard.setAttribute(new AttributeImpl(Metacard.TAGS, tags));
+
+    return metacard;
+  }
+
+  private Metacard findMetacardForFeature(SimpleFeature feature) throws FeatureIndexingException {
+    String countryCode = (String) feature.getAttribute("ISO_A3");
+
+    QueryRequest queryRequest =
+        new QueryRequestImpl(new QueryImpl(getFilterForCountryCode(countryCode)));
+    try {
+      SourceResponse response = catalogFramework.query(queryRequest);
+      if (response.getResults().isEmpty()) {
+        return null;
+      }
+      return response.getResults().get(0).getMetacard();
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private void removeExistingMetacards() throws FeatureIndexingException {
+    QueryRequest queryRequest = new QueryRequestImpl(new QueryImpl(getFilter()));
+
+    try {
+      security.runWithSubjectOrElevate(
+          () -> {
+            SourceResponse response = catalogFramework.query(queryRequest);
+            for (Result result : response.getResults()) {
+              catalogFramework.delete(new DeleteRequestImpl(result.getMetacard().getId()));
+            }
+            return null;
+          });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      throw new FeatureIndexingException(e.getMessage());
+    }
+  }
+
+  private Filter getFilterForCountryCode(String countryCode) {
+    List<Filter> filters = new ArrayList<>();
+    filters.add(filterBuilder.attribute(Metacard.TAGS).is().like().text("gazetteer"));
```

Comment:
Can we make `filterBuilder.attribute(Metacard.TAGS).is().like().text("gazetteer")` into a global?
Can we make `filterBuilder.attribute(Metacard.TAGS).is().like().text("country")` into a global?

Reply:
:+1: need to make globals for the tag values too